source/uwp/Renderer/lib/AdaptiveColumnSet.cpp

```diff
@@ -158,11 +84,11 @@ namespace AdaptiveCards { namespace Rendering { namespace Uwp
 
         XamlHelpers::IterateOverVector<IAdaptiveColumn>(m_columns.Get(), [&](IAdaptiveColumn* column)
         {
-            std::shared_ptr<Column> sharedColumn = std::make_shared<Column>();
+            std::shared_ptr<BaseCardElement> sharedColumn = std::make_shared<Column>();
             ComPtr<AdaptiveCards::Rendering::Uwp::AdaptiveColumn> columnImpl = PeekInnards<AdaptiveCards::Rendering::Uwp::AdaptiveColumn>(column);
             RETURN_IF_FAILED(columnImpl->GetSharedModel(sharedColumn));
 
-            columnSet->GetColumns().push_back(sharedColumn);
+            columnSet->GetColumns().push_back(std::dynamic_pointer_cast<AdaptiveCards::Column>(sharedColumn));
```

Comment:
>dynamic_pointer_cast [](start = 51, length = 20)

Should this one be a static_pointer_cast like the other ones being changed?

Reply:
I made this one dynamic cast since we're down casting so theoretically we can't be sure that the basecardelement is a column. 

On the flip side, we know the implementation of GetSharedModel so we know it would be safe to static cast, and the build system in windows doesn't like dynamic casts.

I'm going to leave as dynamic for now, but i am going to revisit the Windows guidance on this and see whether we should be making a change all up.

---
In reply to: [173899229](https://github.com/Microsoft/AdaptiveCards/pull/1247#discussion_r173899229) [](ancestors = 173899229)

Reply:
(I will add a check on the return value though)

---
In reply to: [174513509](https://github.com/Microsoft/AdaptiveCards/pull/1247#discussion_r174513509) [](ancestors = 174513509,173899229)